sub-packages/bionemo-testing/tests/bionemo/testing/test_megatron_parallel_state_utils.py

```diff
@@ -106,3 +110,47 @@ def test_reduce_scatter():
 
         dist.reduce_scatter(output_tensor, to_reduce_scatter)
         assert tuple(output_tensor.shape) == (3, 3)
+
+
+# def test_all_reduce_sum():
+#     with megatron_parallel_state_utils.mock_distributed_parallel_state(world_size=2, rank=1):
+#         tensor = torch.tensor([dist.get_rank()+1])
+#         dist.all_reduce(tensor)
+#         assert tensor.item() == (1+2) / 2  # TODO does not work; there is no barrier for the actual communication; got 2
+
+
+# move to src
+@contextmanager
+def dist_environment(
+    world_size: int = 1,
+    rank: int = 1,
+):
+    with MonkeyPatch.context() as context:
+        torch.cuda.empty_cache()
+        parallel_state.destroy_model_parallel()
+
+        context.setenv("MASTER_ADDR", "localhost")
+        context.setenv("MASTER_PORT", "29500")
+        dist.init_process_group(backend="nccl", world_size=world_size, rank=rank)
+        yield
+        dist.destroy_process_group()
+        torch.cuda.empty_cache()
+        parallel_state.destroy_model_parallel()
+
+
+def _test_all_reduce_sum(rank: int, world_size: int):
+    with dist_environment(rank=rank, world_size=world_size):
```

Comment:
Would be nice to eventually not need any kind of mocking etc, and have some kind of vanilla comparison of results through some kind of standard lightning etc function that's distributed with different levels of different kinds of parallelism asserting that results are equal to the base case (DDP only).